src/components/WeatherIcon.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { View, Image, Text, StyleSheet } from 'react-native';
+
+enum Weathertype {
+    Sunny,
+    Thunderstorm,
+}
+
+export function WeatherIcon(type: Weathertype) {
+    switch (type) {
+        case Weathertype.Sunny:
+            return (
+                <View>
+                    <Image source={require('@assets/icons/sun.png')} style={styles.sun} />
+                </View>
+            );
+            break;
+        case Weathertype.Thunderstorm:
+            return (
+                <View>
+                    <Image source={require('@assets/icons/thunderstorm.png')} style={styles.thunderstorm} />
+                </View>
+            );
+            break;
+        default:
+            return <Text>Missed</Text>;
+    }
```

Comment:
As functions we usually use arrow functions: `const WeatherIcon: React.FC = () => {}` We typify this function as React.FC, because we need to also typify props, so we would know what arguments are required for this component. To do that we add props type to generic of React.FC: `React.FC<PropsType>` In that case we need only one prop: weatherType, so it would look like:
 `const WeatherIcon: React.FC<{weatherType: WeatherType}> = props => {}`
In the end of the file you can write `export` default WeatherIcon`
We need component to return consistent value, i mean we shouldn't care of component appearance where it will be used, so we must return object of the same size everytime, but there, in one case it is a text, in the second it has one size, in the third - another size. So you should set the size of the wrapper view and place image in the center of it. Also it is not a good idea to write the same text everytime you add new image and it's better to split functions logic(I will describe how to do it below). As we need image to have different sizes, we should store source, width, height for different images. I think we can make object with source, width, height fields. To typify it - create interface:
![image](https://user-images.githubusercontent.com/52620425/224130704-3a24980c-637d-4e40-a230-da8309aa0595.png)

Then add interface that connects weather type key with object of IImageData type:
![image](https://user-images.githubusercontent.com/52620425/224130881-939c817f-689c-4ddd-af22-1c4e8da4b017.png)

Finally we can create object describing all weather types:
![image](https://user-images.githubusercontent.com/52620425/224131210-eb063480-11ea-4f0b-8ea4-b82d9f860ec0.png)

And our component would only contain JSX code and necessary variable initialization
![image](https://user-images.githubusercontent.com/52620425/224132597-16c6c016-9d41-4044-9b9b-eeb0f0458ddf.png)